### ISSUE TITLE
[tool_crash] SocketException: Send failed, OS Error: No route to host, errno = 65

### ISSUE THREAD

**Issue by sanatandarshan** [flutter/flutter#157267]
## Command
```
flutter --no-color run --machine --track-widget-creation --device-id=00008120-001C55013601A01E --start-paused --dart-define=flutter.inspector.structuredErrors=true lib/main.dart
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
SocketException: Send failed, OS Error: No route to host, errno = 65
```
#0      _NativeSocket.send (dart:io-patch/socket_patch.dart:1269:34)
#1      _RawDatagramSocket.send (dart:io-patch/socket_patch.dart:2584:15)
#2      MDnsClient.lookup (package:multicast_dns/multicast_dns.dart:225:14)
#3      MDnsVmServiceDiscovery._pollingVmService (package:flutter_tools/src/mdns_discovery.dart:232:66)
<asynchronous suspension>
#4      MDnsVmServiceDiscovery.firstMatchingVmService (package:flutter_tools/src/mdns_discovery.dart:188:56)
<asynchronous suspension>
#5      MDnsVmServiceDiscovery.getVMServiceUriForLaunch (package:flutter_tools/src/mdns_discovery.dart:433:50)
<asynchronous suspension>
#6      Future.any.onValue (dart:async/future.dart:618:5)
<asynchronous suspension>
```
```
[✓] Flutter (Channel stable, 3.22.2, on macOS 15.0.1 24A348 darwin-arm64, locale en-IN)
    • Flutter version 3.22.2 on channel stable at /Users/dreamworld/workspace/flutter-apps/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (5 months ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/dreamworld/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16A242d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio 3.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.85.1)
    • VS Code at /Users/dreamworld/Downloads/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • Gaurav’s iPhone (mobile)        • 00008120-001C55013601A01E • ios            • iOS 18.1 22B5069a
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 15.0.1 24A348 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 15.0.1 24A348 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 129.0.6668.101

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

## Flutter Application Metadata
**Type**: app
**Version**: 42.0.0+42
**Material**: false
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: stable
**Creation framework version**: 300451adae589accbece3490f4396f10bdf15e6e
### Plugins
audioplayers-6.1.0
audioplayers_android-5.0.0
audioplayers_darwin-6.0.0
audioplayers_linux-4.0.0
audioplayers_web-5.0.1
audioplayers_windows-4.0.0
firebase_core-2.32.0
firebase_core_web-2.18.1
firebase_remote_config-4.4.7
firebase_remote_config_web-1.6.7
fluttertoast-8.2.8
google_sign_in-6.2.1
google_sign_in_android-6.1.31
google_sign_in_ios-5.7.8
google_sign_in_web-0.12.4+2
keep_screen_on-3.0.0
package_info-2.0.2
path_provider-2.1.4
path_provider_android-2.2.10
path_provider_foundation-2.4.0
path_provider_linux-2.2.1
path_provider_windows-2.3.0
permission_handler-10.4.5
permission_handler_android-10.3.6
permission_handler_apple-9.1.4
permission_handler_windows-0.1.3
record-5.1.2
record_android-1.2.6
record_darwin-1.1.2
record_linux-0.7.2
record_web-1.1.3
record_windows-1.0.3
shared_preferences-2.3.2
shared_preferences_android-2.3.2
shared_preferences_foundation-2.5.3
shared_preferences_linux-2.4.1
shared_preferences_web-2.4.2
shared_preferences_windows-2.4.1
sqflite-2.3.3+1
url_launcher-6.3.1
url_launcher_android-6.3.9
url_launcher_ios-6.3.1
url_launcher_linux-3.2.0
url_launcher_macos-3.2.1
url_launcher_web-2.3.3
url_launcher_windows-3.1.2
webview_flutter-2.8.0
webview_flutter_android-2.10.4
webview_flutter_wkwebview-2.9.5

**Comment by FusoraTech** [comment]
Or a Sequoia bug to be fixed in the upcoming 15.1 to be released before the end of the month, or some involved app was blocked by the new Sequoia "security feature" that blocks some apps from using local networking. 
Try going to System Settings -> Privacy & Security, and select Local Network. Check if you find some suspicious app involved and blocked by the system.  If finding, toggle the "Allow" option to ON and try again.

**Comment by huycozy** [comment]
Closing this as a duplicate of https://github.com/flutter/flutter/issues/150131.
